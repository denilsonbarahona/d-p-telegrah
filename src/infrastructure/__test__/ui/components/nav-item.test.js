import React from "react";
import NavItem from "../../../ui/components/nav-item";
import { render } from "@testing-library/react";
import RouterMock from "../../../__mocks__/routes/routeMock";

describe("Nav item", () => {
  it("rendering without crashing", () => {
    const { queryByRole } = render(
      <RouterMock>
        <NavItem selected to="/">
          <span>test</span>
        </NavItem>
      </RouterMock>
    );
    const Link = queryByRole("link");
    expect(Link).toBeInTheDocument();
  });

  it("to match snapshot", () => {
    const { container } = render(
      <RouterMock>
        <NavItem selected to="/">
          <span>test</span>
        </NavItem>
      </RouterMock>
    );
    expect(container).toMatchSnapshot();
  });

  it("to have isSelected className", () => {
    const { container } = render(
      <RouterMock>
        <NavItem selected to="/">
          <span>test</span>
        </NavItem>
      </RouterMock>
    );
    expect(
      container.getElementsByClassName("isSelected")[0]
    ).toBeInTheDocument();
  });

  it("to not have a className", () => {
    const { container } = render(
      <RouterMock>
        <NavItem to="/">
          <span>test</span>
        </NavItem>
      </RouterMock>
    );
    expect(container.getElementsByClassName("isSelected").length).toBe(0);
  });
});
