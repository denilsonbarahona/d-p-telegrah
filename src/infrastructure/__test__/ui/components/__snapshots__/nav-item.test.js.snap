// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav item to match snapshot 1`] = `
<div>
  <a
    class="sc-bczRLJ tEisB"
    href="/"
  >
    <div
      class="sc-gsnTZi dFFSPF isSelected"
    >
      <span>
        test
      </span>
    </div>
  </a>
</div>
`;
